api/src/com/cloud/naming/ResourceNamingPolicyManager.java

```diff
@@ -0,0 +1,32 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package com.cloud.naming;
+
+import com.cloud.utils.component.Manager;
+
+public interface ResourceNamingPolicyManager extends Manager {
+
+
+    /**
+     * Finds a resource naming policy among the registered ones by its class
+     * @param policyClass
+     * @return
+     */
+    @SuppressWarnings("rawtypes")
+    public <T extends ResourceNamingPolicy> T getPolicy(Class<T> policyClass);
```

Comment:
That is great we seem to be on the same page ;)
Sorry for just going through this PR now that you have already done a lot of work.

I think instead of using the “getRegistered” to retrieve a bean across modules, we could take advantage of the application context hierarchic that CloudStack already has.

To use it as Spring singletons and to plan the changes, we would need to know in which modules those objects will be needed. CloudStack already has a module hierarchic, so it would be a matter mapping the modules that will need those objects and them finding a nice spot to put those objects. For that we might even create another module (not sure yet). First things first, do you have a mapping of the modules/objects that need those policies? Something visual would be great :)